Safe_softmax demonstration (#263) (#1950)
huggingface/optimum-habana: docs/source/usage_guides/safe_softmax.mdx

# Comparing HPU-Optimized `safe_softmax` with Native PyTorch `safe_softmax`

> This article demonstrates how to benchmark and compare the performance of the Habana Processing Unit (HPU)-optimized `safe_softmax` operation against the native PyTorch implementation. The provided Python script walks through the process step-by-step, and we will explain each part in detail. Additionally, we will provide some context about `safe_softmax`, its purpose, and its use cases.

Reviewer: ```suggestion
This article demonstrates how to benchmark and compare the performance of the Habana Processing Unit (HPU)-optimized `safe_softmax` operation against the native PyTorch implementation. The provided Python script guides you through the process step-by-step, with detailed explanations for each part. Additionally, we will provide some context about `safe_softmax`, its purpose, and its use cases.
```

---

## Important Note: No Special Setup Required

> The `safe_softmax` operation works out-of-the-box in PyTorch. If you are running your code on Habana hardware, the HPU-optimized implementation will automatically be used without requiring any additional configuration. This seamless integration ensures that you can benefit from the performance improvements without modifying your existing code.

Reviewer: ```suggestion
The `safe_softmax` operation works out-of-the-box in PyTorch. When running your code on Habana hardware, the HPU-optimized implementation is automatically utilized without any additional configuration. This seamless integration allows you to benefit from performance improvements without modifying your existing code.
```
